.github/workflows/build_deploy_rcb_fe.yml

```diff
@@ -0,0 +1,179 @@
+name: Build & Deploy RecipeBox Frontend
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - recipe-box-frontend/**
+      - .github/workflows/build_deploy_rcb_fe.yml
+  pull_request:
+    branches:
+      - main
+    paths:
+      - recipe-box-frontend/**
+      - .github/workflows/build_deploy_rcb_fe.yml
+
+# Azure login
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download and setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Download and install maven
+        run: sudo apt -qq update & sudo apt -qq install maven -y
+
+      - name: Build web app using maven
+        run: mvn install -B -q
+        working-directory: recipe-box-frontend
```

Comment:
Aici o sa vrem sa putem suprascrie o variabila in frontend care sa duca la masina de backend(care este aceeasi de fapt, dar pentru client browser nu o sa stie asta), respectiv domain-ul daca folosim cloudflare sau ip-ul